upup/pkg/fi/cloudup/openstack/cloud.go

```diff
@@ -494,8 +511,34 @@ func (c *openstackCloud) DNS() (dnsprovider.Interface, error) {
 	return provider, nil
 }
 
+// FindVPCInfo list subnets in network
 func (c *openstackCloud) FindVPCInfo(id string) (*fi.VPCInfo, error) {
-	return nil, fmt.Errorf("openstackCloud::FindVPCInfo not implemented")
+	vpcInfo := &fi.VPCInfo{}
+	// Find subnets in the network
+	{
+		if len(c.zones) == 0 {
+			return nil, fmt.Errorf("Could not initialize zones")
+		}
+		klog.V(2).Infof("Calling ListSubnets for subnets in Network %q", id)
+		opt := subnets.ListOpts{
+			NetworkID: id,
+		}
+		subnets, err := c.ListSubnets(opt)
+		if err != nil {
+			return nil, fmt.Errorf("error listing subnets in network %q: %v", id, err)
+		}
+
+		for index, subnet := range subnets {
+			zone := c.zones[int(index)%len(c.zones)]
```

Comment:
This seems fragile - should `c.zones` be a map?

Reply:
this is similar logic that we have in https://github.com/kubernetes/kops/blob/master/pkg/model/openstackmodel/servergroup.go#L99 or https://github.com/kubernetes/kops/blob/master/pkg/model/openstackmodel/servergroup.go#L109

Reply:
> via tags?).



> Another option is to annotate the zones onto the subnets

This is a great idea however some of the openstack resources do not have tag support yet.
https://godoc.org/github.com/gophercloud/gophercloud/openstack/networking/v2/subnets#CreateOpts